pkg/providers/aws/provider_redis.go

```diff
@@ -346,6 +372,75 @@ func (p *RedisProvider) createElasticacheCluster(ctx context.Context, r *v1alpha
 	return &providers.RedisCluster{DeploymentDetails: rdd}, croType.StatusMessage(fmt.Sprintf("successfully created and tagged, aws elasticache status is %s", *foundCache.Status)), nil
 }
 
+func (p *RedisProvider) checkAndApplyMaintenanceUpdates(r *v1alpha1.Redis, cacheSvc elasticacheiface.ElastiCacheAPI, ec2Svc ec2iface.EC2API, elasticacheConfig *elasticache.CreateReplicationGroupInput, serviceUpdates *ServiceUpdate) (croType.StatusMessage, error) {
+	logger := p.Logger.WithField("action", "checkAndApplyMaintenanceUpdates")
+	rgs, err := getReplicationGroups(cacheSvc)
+	if err != nil {
+		// return error so this function can be requeueed
+		errMsg := "error getting replication groups"
+		logger.Info(errMsg, err)
+		return croType.StatusMessage(errMsg), errorUtil.Wrapf(err, errMsg)
+	}
+	// check if the cluster has already been created
+	var foundCache *elasticache.ReplicationGroup
+	for _, c := range rgs {
+		if *c.ReplicationGroupId == *elasticacheConfig.ReplicationGroupId {
+			foundCache = c
+			break
+		}
+	}
+
+	if foundCache != nil {
+		cacheClustersOutput, err := cacheSvc.DescribeCacheClusters(&elasticache.DescribeCacheClustersInput{})
+		if err != nil {
+			errMsg := "failed to describe clusters"
+			return croType.StatusMessage(errMsg), errorUtil.Wrapf(err, errMsg)
+		}
+
+		var replicationGroupClusters []elasticache.CacheCluster
+		for _, checkedCluster := range cacheClustersOutput.CacheClusters {
+			cluster := *checkedCluster
+			if resources.SafeStringDereference(cluster.ReplicationGroupId) == *foundCache.ReplicationGroupId {
+				replicationGroupClusters = append(replicationGroupClusters, *checkedCluster)
+
+				if checkedCluster.EngineVersion != nil && r.Status.Version != *checkedCluster.EngineVersion {
+					r.Status.Version = *checkedCluster.EngineVersion
+				}
+			}
+		}
+
```

Comment:
I think this would only be possible if the replication group existed without any cache clusters, which I'm not sure is possible. This logic has been used on L310 during provisioning of redis for the last 2 years so I think we might be safe